amethyst_renderer/src/sprite.rs

```diff
@@ -24,19 +27,38 @@ pub struct SpriteSheet {
 }
 
 impl Asset for SpriteSheet {
-    const NAME: &'static str = "renderer::Sprite";
+    const NAME: &'static str = "renderer::SpriteSheet";
     type Data = Self;
     type HandleStorage = VecStorage<Handle<Self>>;
 }
 
-/// A description of a frame in a sprite sheet.
+impl From<SpriteSheet> for AssetsResult<ProcessingState<SpriteSheet>> {
+    fn from(sprite_sheet: SpriteSheet) -> AssetsResult<ProcessingState<SpriteSheet>> {
+        Ok(ProcessingState::Loaded(sprite_sheet))
+    }
+}
+
+/// Dimensions and texture coordinates of each sprite in a sprite sheet.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Sprite {
+    /// Pixel width of the sprite
+    pub width: f32,
+    /// Pixel height of the sprite
+    pub height: f32,
+    /// Number of pixels to shift the sprite to the left and down relative to the entity
+    pub offsets: [f32; 2],
+    /// Texture coordinates of the sprite
+    pub tex_coords: TextureCoordinates,
+}
+
+/// Texture coordinates of the sprite
 ///
-/// These should be in normalized coordinates:
+/// The coordinates should be normalized to a value between 0.0 and 1.0:
 ///
 /// * X axis: 0.0 is the left side and 1.0 is the right side.
 /// * Y axis: 0.0 is the top and 1.0 is the bottom.
```

Comment:
I don't quite remember, but I think we settled on using a bottom to up Y axis?

Reply:
(even the ui uses Y up to avoid confusion)

Reply:
I know it should be bottom up for world coordinates, I wasn't sure about sprites, but I can make this change

* [ ] bottom up texture Y coordinates

Reply:
thanks

Reply:
ui doesn't use Y = up, but I thought it did :P
anyway, sprites should be bottom up still